packages/core/test/endowmentsToolkit.spec.js

```diff
@@ -45,6 +45,57 @@ test('getEndowmentsForConfig - function on proto', (t) => {
   t.is(resultGlobal.lookAtMyProto.appendChild(), assertMe)
 })
 
+test('getEndowmentsForConfig - function call/bind unwrapping', (t) => {
+  const getEndowmentsForConfig = prepareTest()
+
+  const sourceGlobal = {
+    checkThis: function () {
+      return this === sourceGlobal
+    },
+  }
+  const config = {
+    globals: {
+      checkThis: true,
+    },
+  }
+  const resultGlobal = getEndowmentsForConfig(sourceGlobal, config)
+  t.is(typeof resultGlobal.checkThis, 'function')
+  t.is(resultGlobal.checkThis(), true)
+  t.is(resultGlobal.checkThis.bind({})(), false)
+  t.is(resultGlobal.checkThis.bind(resultGlobal)(), true)
+  t.is(resultGlobal.checkThis.call(resultGlobal), true)
+})
+
+test.failing(
+  'getEndowmentsForConfig - function call/bind unwrapping when only methods endowed',
+  (t) => {
+    const getEndowmentsForConfig = prepareTest()
+
+    const sourceGlobal = {
+      checkThis: function () {
+        return this === sourceGlobal
+      },
+    }
+
+    const config = {
+      globals: {
+        'checkThis.bind': true,
+        'checkThis.call': true,
+      },
+    }
+    const resultGlobal = getEndowmentsForConfig(sourceGlobal, config)
+    t.is(typeof resultGlobal.checkThis, 'object') // the function is never wrapped
+
+    // This case is not supported and might be too risky to support. We should rather avoid producing it.
+    // In policy generation we could trim .bind (and probably also .call and .apply) to avoid hitting this problem.
```

Comment:
I'd like to better understand what's risky here.

(please fix) I'd recommend splitting this up into two tests, one for each assertion; when the first assertion fails (as it presumably should), the test should exit and the second assertion will never be called. This assumes AVA's behavior is not completely wacko.

Reply:
It'd be really hard to avoid creating a way to leak a reference that the script should not have according to policy while trying to get the binding to work correctly.

Reply:
can you change the test though? 😄